src/test/java/core/basesyntax/service/impl/FruitTransactionParserImplTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.model.FruitTransaction.Operation.BALANCE;
+import static core.basesyntax.model.FruitTransaction.Operation.PURCHASE;
+import static core.basesyntax.model.FruitTransaction.Operation.RETURN;
+import static core.basesyntax.model.FruitTransaction.Operation.SUPPLY;
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionParser;
+import core.basesyntax.service.OperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionParserImplTest {
+    private static final String KEY_APPLE = "apple";
+    private static final String KEY_BANANA = "banana";
+    private static FruitDao fruitDao;
+    private static FruitTransactionParser fruitTransactionParser;
+    private static List<FruitTransaction> actualFruitTransactions;
+    private static Map<FruitTransaction.Operation, OperationHandler> handlerMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fruitTransactionParser = new FruitTransactionParserImpl();
+        fruitDao = new FruitDaoImpl();
+        handlerMap = new HashMap<>();
+        handlerMap.put(BALANCE, new BalanceOperationHandlerImpl(fruitDao));
+        handlerMap.put(SUPPLY, new AddOperationHandlerImpl(fruitDao));
+        handlerMap.put(PURCHASE, new SubtractOperationHandlerImpl(fruitDao));
+        handlerMap.put(RETURN, new ReturnOperationHandlerImpl(fruitDao));
+        actualFruitTransactions = new ArrayList<>(List.of(
+                new FruitTransaction(BALANCE, KEY_BANANA,200),
+                new FruitTransaction(SUPPLY, KEY_APPLE,100),
+                new FruitTransaction(PURCHASE, KEY_BANANA,100),
+                new FruitTransaction(RETURN, KEY_APPLE,0)));
```

Comment:
let's make `actualFruitTransactions` a local variable and move to `parseData_IsOk`